### DISCUSSION TITLE
SingleStrip TIFF

### DISCUSSION THREAD

**Discussion by tdbrown1** [SixLabors/ImageSharp#2502] [Q&A]
I need to be able to produce a FED TIFF image that requires 1 bit per pixel and single strip.  I dont see how to do that with this library.

I saw a PR to do this but it isn't in the latest build.
https://github.com/SixLabors/ImageSharp/pull/1537

Maybe a simple change would be to do something like this:
**Please don't take this as an exact solution.**

```
public enum StripsPerRow { Auto, Multiple, Single }
public class TiffEncoder : QuantizingImageEncoder { public StripsPerRow StripsPerRow  { get; init; } = StripsPerRow.Auto; ... }


internal sealed class TiffEncoderCore : IImageEncoderInternals {
...
    private readonly StripsPerRow  stripsPerRow;
   public TiffEncoderCore(TiffEncoder options, MemoryAllocator memoryAllocator)
   {
      ...
      stripsPerRow = options.StripsPerRow;
   }
    private static int CalcRowsPerStrip(int height, int bytesPerRow, TiffCompression? compression)
    {
        DebugGuard.MustBeGreaterThan(height, 0, nameof(height));
        DebugGuard.MustBeGreaterThan(bytesPerRow, 0, nameof(bytesPerRow));

       if(stripsPerRow == StripsPerRow.Single)
         return 1;

...
        return 1;
    }
```

**Comment by tdbrown1** [comment]
Fun

**Reply by JimBobSquarePants** [reply]
That PR was merged over 2 years ago. There's been multiple builds including a major since.

**Reply by tdbrown1** [reply]
I cannot find it in the library.  Is it in an extension library?

**Reply by JimBobSquarePants** [reply]
No, the way it was calculated was changed. It's based upon the `TiffCompression` value. Only `TiffCompression.Jpeg` uses a single strip at the moment.

@brianpopow @IldarKhayrutdinov You guys are both much better with the specification than me. Is the suggestion above reasonable?

**Reply by IldarKhayrutdinov** [reply]
@JimBobSquarePants Right.

@tdbrown1 If you want one strip to consist one image row, at the moment it's auto calculating, only jpeg is used RowsPerStrip = 1.

https://github.com/SixLabors/ImageSharp/blob/77d49e6732402ef1b4bce0603b6fb9adbc729d98/src/ImageSharp/Formats/Tiff/TiffEncoderCore.cs#L262-L288
 
https://github.com/SixLabors/ImageSharp/blob/77d49e6732402ef1b4bce0603b6fb9adbc729d98/src/ImageSharp/Formats/Tiff/Writers/TiffBaseColorWriter%7BTPixel%7D.cs#L32

https://github.com/SixLabors/ImageSharp/blob/77d49e6732402ef1b4bce0603b6fb9adbc729d98/src/ImageSharp/Formats/Tiff/Constants/TiffConstants.cs#L59

RowsPerStrip = 1 - only if row bytes doesn't fit into `DefaultStripSize * 2` or `DefaultStripSize`.
